packages/monorepo-tools/src/utils/get-packages-in-topological-order.spec.ts

```diff
@@ -0,0 +1,264 @@
+import { execFile as execFileCb } from 'child_process';
+import { promisify } from 'util';
+import path from 'path';
+import { promises as fs } from 'fs';
+import os from 'os';
+import { getPackagesInTopologicalOrder } from './get-packages-in-topological-order';
+import assert from 'assert';
+
+const execFile = promisify(execFileCb);
+
+describe('getPackagesInTopologicalOrder', function () {
+  let tempDir: string;
+  let remoteDir;
+  let repoPath: string;
+
+  const writeRepoFile = async (filePath: string, content: any) => {
+    await fs.mkdir(path.dirname(filePath), { recursive: true });
+    await fs.writeFile(
+      path.join(repoPath, filePath),
+      typeof content === 'string' ? content : JSON.stringify(content, null, 2)
+    );
+  };
+
+  beforeEach(async function () {
+    tempDir = await fs.mkdtemp(path.join(os.tmpdir(), 'monorepo-tests-'));
+
+    // create fake git remote:
+    remoteDir = path.resolve(tempDir, 'remote');
+    await fs.mkdir(remoteDir, { recursive: true });
+    process.chdir(remoteDir);
+    await execFile('git', ['init', '--bare']);
+    await execFile('git', ['config', '--local', 'user.name', 'user']);
+    await execFile('git', [
+      'config',
+      '--local',
+      'user.email',
+      'user@example.com',
+    ]);
+
+    // setup repo and package:
+    repoPath = path.resolve(tempDir, 'monorepo-test-repo');
+    await fs.mkdir(repoPath, { recursive: true });
+    process.chdir(repoPath);
```

Comment:
Should this be reset to its original value somewhere so it doesn’t mess with other tests (or even other test files) potentially added in the future?

(I’d personally try to avoid `chdir()` as much as possible, I don’t think having a `let cwd = remoteDir` at the start here and then doing `await execFile(..., { cwd });` each time would be that bad)

Reply:
sure! i've copy-pasted this from other compass tests and probably forgot few lines

Reply:
fun how many bugs i've fixed by switching to { cwd }